### PR TITLE
Update test plan

### PR DESCRIPTION
**Pull request by rudream**
## Purpose

Adds test plan items for testing the new MFA device managment dashboard and Recovery dashboard (for cloud users)

### CODE REVIEW

File: docs/testplan.md [hunk 1]
```diff
@@ -564,24 +564,35 @@ authentication:
 - [ ] Change `second_factor` type to `on` and verify that mfa is required (no option `none` in dropdown)
 
 #### MFA require auth
-Through the CLI, `tsh login` and register a WebAuthn key with `tsh mfa add` (not supported in UI yet).
+Through the CLI, `tsh login` and register a WebAuthn key with `tsh mfa add`.
```
Comment (kimlisa) [inline docs/testplan.md:567]: we don't need to use the CLI anymore for this so lets rewrite to a general `register a webauthn key`

File: docs/testplan.md [hunk 2]
```diff
@@ -564,24 +564,35 @@ authentication:
...
 
 Using the same user as above:
 - [ ] Verify logging in with registered WebAuthn key works
 - [ ] Verify connecting to a ssh node prompts you to tap your registered WebAuthn key
 - [ ] Verify in the web terminal, you can scp upload/download files
 
-#### MFA Management
-TODO(yassine)
+#### MFA Management (UI)
```
Comment (kimlisa) [inline docs/testplan.md:574]:
```suggestion
#### MFA Management
```

this section (and below) is actually under a huge section called [Web UI](https://github.com/gravitational/teleport/blob/master/docs/testplan.md#web-ui), so all test under it implies it's for UI

File: docs/testplan.md [hunk 3]
```diff
@@ -564,24 +564,35 @@ authentication:
...
 
 ## Cloud
 From your cloud staging account, change the field `teleportVersion` to the test version.
 ```
 $ kubectl -n <namespace> edit tenant
 ```
 
-#### Recovery Code Management
-TODO(yassine)
+#### Recovery Code Management (UI)
```
Comment (kimlisa) [inline docs/testplan.md:591]:
```suggestion
#### Recovery Code Management
```

File: docs/testplan.md [hunk 4]
```diff
@@ -564,24 +564,35 @@ authentication:
...
+#### MFA Management (UI)
+
+- [ ] Verify adding first device works without requiring re-authentication
+- [ ] Verify re-authenticating with a WebAuthn device works
+- [ ] Verify re-authenticating with a U2F device works
+- [ ] Verify re-authenticating with a OTP device works
+- [ ] Verify adding a WebAuthn device works
+- [ ] Verify adding a U2F device works
+- [ ] Verify adding an OTP device works
+- [ ] Verify removing a device works
```
Comment (kimlisa) [inline docs/testplan.md:583]: lets also add
```
- [ ] Verify `second_factor` set to `off` disables adding devices
```

File: docs/testplan.md [hunk 5]
```diff
@@ -564,24 +564,35 @@ authentication:
...
 ```
 $ kubectl -n <namespace> edit tenant
 ```
 
-#### Recovery Code Management
-TODO(yassine)
+#### Recovery Code Management (UI)
+
+- [ ] Verify generating recovery codes for local accounts with email usernames works
+- [ ] Verify local accounts with non-email usernames don't have access to generate recovery codes
```
Comment (kimlisa) [inline docs/testplan.md:594]:
```suggestion
- [ ] Verify local accounts with non-email usernames are not able to generate recovery codes
```

File: docs/testplan.md [hunk 6]
```diff
@@ -564,24 +564,35 @@ authentication:
...
+- [ ] Verify SSO accounts don't have access to generate recovery codes
```
Comment (kimlisa) [inline docs/testplan.md:595]:
```suggestion
- [ ] Verify SSO accounts are not able to generate recovery codes
```

**Review by kimlisa — DISMISSED** [review]
After merge, lets copy and paste these additions into our [8.0 test plan](https://github.com/gravitational/teleport/issues/8665). We will then go through it after we finish backporting to `branch/v8` and updating webassets

File: docs/testplan.md [hunk 7]
```diff
@@ -564,15 +564,24 @@ authentication:
 - [ ] Change `second_factor` type to `on` and verify that mfa is required (no option `none` in dropdown)
 
 #### MFA require auth
-Through the CLI, `tsh login` and register a WebAuthn key with `tsh mfa add` (not supported in UI yet).
+Go to Account Settings > Two-Factor Devices and register a new device by clicking "Add new two-factor device"
```
Comment (kimlisa) [inline docs/testplan.md:567]:
```suggestion
Go to `Account Settings` > `Two-Factor Devices` and register a new device
```

**Review by kimlisa — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Pull request merged**